From d49bec0c81d958e81739366c48f21d493ebb746d Mon Sep 17 00:00:00 2001
From: Zbigniew Lukwinski <zbigniew.lukwinski@intel.com>
Date: Mon, 19 May 2025 12:14:42 +0200
Subject: [PATCH] EphemeralPrivateKey from bytes

Added API to create EphemeralPrivateKey from bytes array. This allows to
create the key based on the raw bytes, which is useful for testing
purposes.

Signed-off-by: Zbigniew Lukwinski <zbigniew.lukwinski@intel.com>
---
 src/agreement.rs | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/agreement.rs b/src/agreement.rs
index 770423f2b..a17aebc2c 100644
--- a/src/agreement.rs
+++ b/src/agreement.rs
@@ -153,6 +153,15 @@ impl EphemeralPrivateKey {
     pub(super) fn bytes_for_test(&self) -> &[u8] {
         self.private_key.bytes_less_safe()
     }
+
+    pub fn from_bytes_for_test(alg: &'static Algorithm, bytes: &[u8]) -> Result<Self, error::Unspecified> {
+        let private_key =
+            ec::Seed::from_bytes(alg.curve, untrusted::Input::from(bytes), cpu::features())?;
+        Ok(EphemeralPrivateKey {
+            private_key,
+            algorithm: alg,
+        })
+    }
 }
 
 /// A public key for key agreement.
-- 
2.34.1

